File: zzz_legacy/EFR/EFR_2016-09/EFR/VectorInformation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EFR
{
    public class VectorInformation
    {
        public long VectorID;
        public string Vector;
        public string Support;
        public int SupportSize;
        public string ExtendedSupport;
        public int ExtendedSupportSize;
        public int IsEss;
        public int ReasonEss;
		public BigRational Payoff;
        public decimal PayoffDecimal;

		public string GetVectorInformation()
		{
			StringBuilder str = new StringBuilder("");
			str.Append("'" + this.VectorID + "',"); 
			str.Append("'" + this.Vector + "',"); 
			str.Append("'" + this.Support + "',");
			str.Append("'" + this.SupportSize + "',");
			str.Append("'" + this.ExtendedSupport + "',");
			str.Append("'" + this.ExtendedSupportSize + "',");
			str.Append("'" + this.IsEss + "',");
			str.Append("'" + this.ReasonEss + "',");
			str.Append("'" + this.Payoff.ToString() + "',");
			str.Append("'" + this.PayoffDecimal.ToString().Replace(",", ".") + "'");
			return(str.ToString ());

		}


		public static string GetHeader()
		{
			StringBuilder str = new StringBuilder ("");
			str.Append ("'VectorID',");
			str.Append ("'Vector',");
			str.Append ("'Support',");
			str.Append ("'SupportSize',");
			str.Append ("'ExtendedSupport',");
			str.Append ("'ExtendedSupportSize',");
			str.Append ("'IsEss',");
			str.Append ("'Reason',");
			str.Append ("'Payoff',");
			str.Append ("'PayoffDecimal'");
			return(str.ToString ());
		}
    }
}

File: zzz_legacy/REF_2016-10-06/src/VectorInformation.cpp
#include "VectorInformation.h"

VectorInformation::VectorInformation()
{
    //ctor
}

VectorInformation::~VectorInformation()
{
    //dtor
}

std::string VectorInformation::getVectorInformation()
{
    std::string str ="";
    str+= "'" + std::to_string(vectorID) + "','";
    for (auto x : resultVector)
        str+= x.get_str()+",";
    str.pop_back();
    str+= "','" + std::to_string(support) + "',";
    str+= "'" + std::to_string(supportSize) + "',";
    str+= "'" + std::to_string(extendedSupport) + "',";
    str+= "'" + std::to_string(extendedSupportSize) + "',";
    str+= "'" + std::to_string(shiftReference) + "',";
    str+= "'" + std::to_string(IsEss) + "',";
    str+= "'" + std::to_string(ReasonEss) + "',";
    str+= "'" + payoff.get_str() + "',";
    str+= "'" + std::to_string(payoffDouble) + "',";

    return str;

}


std::string VectorInformation::getHeader()
{
    std::string str ="";
    str+= "'VectorID',";
    str+= "'Vector',";
    str+= "'Support',";
    str+= "'SupportSize',";
    str+= "'ExtendedSupport',";
    str+= "'ExtendedSupportSize',";
    str+= "'ShiftReference',";
    str+= "'IsEss',";
    str+= "'Reason',";
    str+= "'Payoff',";
    str+= "'PayoffDecimal'";

    return str;

}
